### PR TITLE
Move tests and demos to Gradle

### PR DESCRIPTION
**Pull request by ronocod**
Makes all projects and tests usable in Android Studio with Gradle.

* Changes directory structure of the project to use modern Android studio structure, i.e. `src/main/java`
* Moves 2 demo projects to use Gradle
* Delete `acceptance` module and move its tests into `core` module
* Use modern Robolectric version for `core` unit tests

### CODE REVIEW

File: core/src/test/java/novoda/lib/sqliteprovider/util/ShadowDatabaseUtils.java [hunk 1]
```diff
@@ -1,28 +0,0 @@
-package novoda.lib.sqliteprovider.util;
-
-import android.database.DatabaseUtils;
-
-import com.xtremelabs.robolectric.internal.Implementation;
-import com.xtremelabs.robolectric.internal.Implements;
-
-@Implements(DatabaseUtils.class)
-public class ShadowDatabaseUtils extends com.xtremelabs.robolectric.shadows.ShadowDatabaseUtils {
```
Comment (ronocod) [inline core/src/test/java/novoda/lib/sqliteprovider/util/ShadowDatabaseUtils.java]: Not needed any more, we can use the stock Robolectric `ShadowDatabaseUtils` class

File: core/src/test/resources/assets/sql/missing_last_semicolon.sql [hunk 2]
```diff
@@ -1,10 +0,0 @@
--- We ommit the trailing semicolon of the last statement to test whether the parser detects that.
-CREATE TABLE 'testTable'
```
Comment (ronocod) [inline core/src/test/resources/assets/sql/missing_last_semicolon.sql]: Moved these from files to Strings as they're small and we had issues getting the tests to read these files

**Comment by stefanhoth** [comment]
Good job!
On May 1, 2015 12:38 PM, "ronocod" <notifications@github.com> wrote:

> Makes all projects and tests usable in Android Studio with Gradle.
>
>    - Changes directory structure of the project to use modern Android
>    studio structure, i.e. src/main/java
>    - Moves 2 demo projects to use Gradle
>    - Delete acceptance module and move its tests into core module
>    - Use modern Robolectric version for core unit tests
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/novoda/sqlite-provider/pull/48
> Commit Summary
>
>    - Move files into proper Gradle structure
>    - Fix file locations
>    - Move acceptance module tests into core module androidTest folder
>    - Move assets folders to correct locations
>    - Simple demo has no tests
>    - Extended demo has no tests
>    - Remove unneeded dependencies
>    - Make Robolectric tests work
>    - Move SQL statements from files to strings
>    - Lower minSdkVersion to 8
>
> File Changes
>
>    - *D* acceptance/lint.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-0> (9)
>    - *D* acceptance/pom.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-1> (80)
>    - *D* acceptance/proguard-project.txt
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-2> (20)
>    - *D* acceptance/project.properties
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-3> (14)
>    - *D* acceptance/test/created_to_please_maven
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-4> (0)
>    - *M* core/build.gradle
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-5> (20)
>    - *R* core/src/androidTest/AndroidManifest.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-6> (8)
>    - *R* core/src/androidTest/assets/migrations/129467607_create.sql
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-7> (0)
>    - *R* core/src/androidTest/assets/sql/create.sql
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-8> (0)
>    - *R*
>    core/src/androidTest/java/novoda/lib/sqliteprovider/ContentProviderTest.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-9> (0)
>    - *R*
>    core/src/androidTest/java/novoda/lib/sqliteprovider/sqlite/AssetHelperTest.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-10> (0)
>    - *R*
>    core/src/androidTest/java/novoda/lib/sqliteprovider/sqlite/ExtendedSQLiteOpenHelperTest.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-11> (0)
>    - *R*
>    core/src/androidTest/java/novoda/lib/sqliteprovider/sqlite/InsertHelperTest.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-12> (0)
>    - *R*
>    core/src/androidTest/java/novoda/lib/sqliteprovider/testproviders/TestProvider.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-13> (0)
>    - *R*
>    core/src/androidTest/java/novoda/lib/sqliteprovider/util/DBUtilsTest.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-14> (8)
>    - *R* core/src/androidTest/res/raw/create.sql
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-15> (0)
>    - *R*
>    core/src/androidTest/res/raw/inserts_parent_no_tansaction_1000_rows.sql
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-16> (0)
>    - *R* core/src/main/AndroidManifest.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-17> (4)
>    - *M*
>    core/src/test/java/novoda/lib/sqliteprovider/migration/MigrationsTest.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-18> (16)
>    - *M*
>    core/src/test/java/novoda/lib/sqliteprovider/provider/SQLiteProviderLocalTest.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-19> (34)
>    - *M* core/src/test/java/novoda/lib/sqliteprovider/util/RoboRunner.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-20> (15)
>    - *M*
>    core/src/test/java/novoda/lib/sqliteprovider/util/SQLFileTest.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-21> (56)
>    - *M*
>    core/src/test/java/novoda/lib/sqliteprovider/util/ShadowAndroidTestCase.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-22> (7)
>    - *D*
>    core/src/test/java/novoda/lib/sqliteprovider/util/ShadowDatabaseUtils.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-23> (28)
>    - *M* core/src/test/resources/AndroidManifest.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-24> (28)
>    - *D* core/src/test/resources/assets/sql/missing_last_semicolon.sql
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-25> (10)
>    - *D* core/src/test/resources/assets/sql/multi_line_statements.sql
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-26> (14)
>    - *D* core/src/test/resources/assets/sql/one_line_statements.sql
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-27> (4)
>    - *A* demo-extended/build.gradle
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-28> (28)
>    - *R* demo-extended/src/main/AndroidManifest.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-29> (4)
>    - *R* demo-extended/src/main/assets/migrations/1_SETUP.SQL
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-30> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/NovodaApplication.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-31> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/domain/Firework.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-32> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/domain/Groups.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-33> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/domain/Shop.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-34> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/domain/UseCaseFactory.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-35> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/domain/UseCaseInfo.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-36> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/loader/FireworkLoader.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-37> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/loader/FireworkSaver.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-38> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/persistance/DatabaseConstants.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-39> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/persistance/DatabaseReader.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-40> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/persistance/DatabaseWriter.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-41> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/persistance/FireworkReader.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-42> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/persistance/FireworkWriter.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-43> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/provider/FireworkProvider.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-44> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/provider/FireworkUriConstants.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-45> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/AddFireworkActivity.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-46> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/FindDistinctFireworksActivity.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-47> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/FindFireworkWithPkActivity.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-48> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/FindFireworksFromOneShopActivity.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-49> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/FindThreeFireworksActivity.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-50> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/FireworkActivity.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-51> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/MainActivity.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-52> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/ViewAllFireworksActivity.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-53> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/ViewShopActivity.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-54> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/WellStockedShopActivity.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-55> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/adapter/FireworkAdapter.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-56> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/base/NovodaActivity.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-57> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/fragment/AddFireworkFragment.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-58> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/fragment/FindFireworkWithPkFragment.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-59> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/fragment/FindFireworksFromOneShopFragment.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-60> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/fragment/ShopFragment.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-61> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/fragment/UriSqlFragment.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-62> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/input/DemoMenu.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-63> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/input/OnPrimaryKeyInputError.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-64> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/util/FromXML.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-65> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/widget/UriSqlView.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-66> (0)
>    - *R*
>    demo-extended/src/main/java/com/novoda/sqliteprovider/demo/util/Log.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-67> (0)
>    - *R* demo-extended/src/main/res/drawable-hdpi/ic_launcher.png
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-68> (0)
>    - *R* demo-extended/src/main/res/drawable-mdpi/ic_launcher.png
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-69> (0)
>    - *R* demo-extended/src/main/res/drawable-xhdpi/ic_launcher.png
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-70> (0)
>    - *R* demo-extended/src/main/res/layout/activity_add_firework.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-71> (0)
>    - *R*
>    demo-extended/src/main/res/layout/activity_find_firework_from_one_shop.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-72> (0)
>    - *R*
>    demo-extended/src/main/res/layout/activity_find_firework_with_pk.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-73> (0)
>    - *R* demo-extended/src/main/res/layout/activity_firework.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-74> (0)
>    - *R* demo-extended/src/main/res/layout/activity_main.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-75> (0)
>    - *R* demo-extended/src/main/res/layout/activity_view_fireworks.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-76> (0)
>    - *R* demo-extended/src/main/res/layout/activity_view_shop.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-77> (0)
>    - *R* demo-extended/src/main/res/layout/fragment_add_firework.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-78> (0)
>    - *R*
>    demo-extended/src/main/res/layout/fragment_find_firework_from_one_shop.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-79> (0)
>    - *R*
>    demo-extended/src/main/res/layout/fragment_find_firework_with_pk.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-80> (0)
>    - *R* demo-extended/src/main/res/layout/fragment_shop.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-81> (0)
>    - *R* demo-extended/src/main/res/layout/fragment_uri_sql.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-82> (0)
>    - *R* demo-extended/src/main/res/layout/view_uri_sql.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-83> (0)
>    - *R* demo-extended/src/main/res/values/colors.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-84> (0)
>    - *R* demo-extended/src/main/res/values/dimens.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-85> (0)
>    - *R* demo-extended/src/main/res/values/layout_styles.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-86> (0)
>    - *R* demo-extended/src/main/res/values/strings.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-87> (0)
>    - *R* demo-extended/src/main/res/values/styles.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-88> (0)
>    - *D* demo-extended/test/created_to_please_maven
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-89> (0)
>    - *A* demo-simple/build.gradle
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-90> (28)
>    - *R* demo-simple/src/main/AndroidManifest.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-91> (4)
>    - *R* demo-simple/src/main/assets/migrations/1_SETUP.SQL
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-92> (0)
>    - *R*
>    demo-simple/src/main/java/com/novoda/sqliteprovider/demo/simple/provider/FireworkProvider.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-93> (0)
>    - *R*
>    demo-simple/src/main/java/com/novoda/sqliteprovider/demo/simple/ui/MainActivity.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-94> (0)
>    - *R*
>    demo-simple/src/main/java/com/novoda/sqliteprovider/demo/simple/ui/MainFragment.java
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-95> (0)
>    - *R* demo-simple/src/main/res/drawable-hdpi/ic_launcher.png
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-96> (0)
>    - *R* demo-simple/src/main/res/drawable-mdpi/ic_launcher.png
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-97> (0)
>    - *R* demo-simple/src/main/res/drawable-xhdpi/ic_launcher.png
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-98> (0)
>    - *R* demo-simple/src/main/res/layout/activity_main.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-99> (0)
>    - *R* demo-simple/src/main/res/layout/fragment_main.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-100> (0)
>    - *R* demo-simple/src/main/res/values/ids.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-101> (0)
>    - *R* demo-simple/src/main/res/values/strings.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-102> (0)
>    - *R* demo-simple/src/main/res/values/styles.xml
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-103> (0)
>    - *D* demo-simple/test/created_to_please_maven
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-104> (0)
>    - *M* settings.gradle
>    <https://github.com/novoda/sqlite-provider/pull/48/files#diff-105> (4)
>
> Patch Links:
>
>    - https://github.com/novoda/sqlite-provider/pull/48.patch
>    - https://github.com/novoda/sqlite-provider/pull/48.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/sqlite-provider/pull/48>.
>

-- 
Disclaimer: The information in this e-mail and any attachments is the 
property of Novoda Ltd and is confidential and may be legally privileged. 
This e-mail is intended solely for the person or organisation to which it 
is addressed. Any disclosure, copying or other use of the information by 
any person or organisation who is not the intended recipient is strictly 
prohibited and may be unlawful. If you have received this e-mail in error, 
please inform the sender immediately and delete/destroy this e-mail and any 
copies of it. Novoda Ltd has taken reasonable precautions to minimise the 
risk of any software viruses which may damage your systems, but we advise 
that you take the necessary steps to ensure that no virus contamination is 
suffered. Novoda Ltd does not accept any liability for any loss or damage 
caused by the transmission of any virus.

Novoda Ltd, Company No: 347444, Registered in Scotland Registered 
Office: C/O Alexander Sloan, 38 Cadogan Street, Glasgow, G2 7HF, Scotland.  VAT 
Registration Number  GB 984 2525 93

**Comment by xrigau** [comment]
Fixes https://github.com/novoda/sqlite-provider/issues/39

**Comment by xrigau** [comment]
Good stuff! :+1: red > green is always good

Will merge if no more comments

File: core/src/test/resources/assets/sql/missing_last_semicolon.sql (hunk 2 above)
Comment (blundell) [inline core/src/test/resources/assets/sql/missing_last_semicolon.sql]: more readable as files though - maybe raise an issue on the repo to revert this when we can

File: demo-extended/build.gradle [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    }
+}
+
+repositories {
+    jcenter()
+}
+
+dependencies {
+    compile project(':core')
+    compile 'com.android.support:appcompat-v7:22.1.1'
```
Comment (blundell) [inline demo-extended/build.gradle]: demo's are places I would use + :speak_no_evil:  `appcompat-v7:22.+`

**Comment by blundell** [comment]
LGTM :+1:

File: core/build.sbt [hunk 4]
```diff
@@ -1,26 +0,0 @@
-import saab.Plugin._
```
Comment (xrigau) [inline core/build.sbt]: @charroch be like

**Pull request merged**